lia-ui/src/main/java/com/lithium/community/android/ui/components/activities/LiSupportHomeActivity.java

```diff
@@ -207,17 +312,60 @@ public void onFocusChange(View view, boolean hasFocus) {
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
         int i = item.getItemId();
-        if (i == android.R.id.home) {
-            finish();
-        } else if (i == R.id.li_action_search) {
+        if (i == R.id.li_action_search) {
             return onSearchRequested();
+        } else if (i == android.R.id.home) {
+            if (drawerLayout.isDrawerOpen(Gravity.START)) {
+                drawerLayout.closeDrawer(Gravity.START);
+            } else {
+                drawerLayout.openDrawer(Gravity.START);
+            }
         }
-
         return super.onOptionsItemSelected(item);
     }
 
     @Override
     public void onConfigurationChanged(Configuration newConfig) {
         super.onConfigurationChanged(newConfig);
     }
+
+    private BroadcastReceiver receiver = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
```

Comment:
Does the lifecycle of the receiver need to be tied to the lifecycle of the activity?

Reply:
No, it doesn't linearly relate to the activity. Because Activity also has a flow which doesn't need Login.